### PR TITLE
Encapsulate execution environment using Docker

### PR DESCRIPTION
**Pull request by jsurls**
Created a Dockerfile and wrapper shell script to make this quickly executable for users without having to think about installation cross platform, and debugging any dependency installs, keeping the setup minimized to just:

- Setup Docker
- Setup AWS

The goal was to help users that are aiming to make custom practice files easier without too much of a developer deep dive keeping up with the dependencies to install and which versions. I took a stab at updating the README but feel free to tweak to how you prefer to communicate it.

Basically, as a user I didn't want to dig too much into "how to setup" and wanted to "just use it" with the least effort to get it up and running :)

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+version: '3'
+services:
+  app:
+    container_name: morse-code-ninja
+    build: .
+    volumes:
+      - .:/opt/morse-code-ninja/
+    environment:
+      - AWS_KEY_ID=${AWS_ACCESS_KEY_ID}
```
Comment (jsurls) [inline docker-compose.yml:9]: I can update this to `AWS_ACCESS_KEY_ID` after https://github.com/zoglmannk/Morse-Code-Ninja/pull/45 is merged.
